do not use tstamp from tl_files as the mtime

fritzmg opened the pull request:
Currently, the cache busting parameter that is automatically appended via `Template::generateScriptTag` will not change for `externalJs` files of the page layout whenever those files change, since the current code uses the `tstamp` from `tl_files` rather than the `mtime` of the actual file. The `tl_files.tstamp` will of course only change if you change something in the file manager in the back end for that file.

### Reproduction

1. Add an external JavaScript file to the page layout.
2. Open the front end and note the `?v=` parameter appended to that file.
3. Change the content of the JavaScript file.
4. Open the front end again.

The `?v=` parameter will still be the same.

fritzmg committed:
do not use tstamp from tl_files as the mtime

fritzmg committed:
also remove the second parameter

leofeyer commented:
Isn't the same logic used here as well?

https://github.com/contao/contao/blob/6ac8629d18c83d446a05ab77c3c8b06920f365bf/core-bundle/src/Resources/contao/pages/PageRegular.php#L562-L567

We should either change it in both places or not at all, shouldn't we?

fritzmg commented:
I guess so. I simply haven't noticed this, since the code you mentioned only applies when:
```php
$objStylesheets->cc || ($objStylesheets->hasFontFace && $media != 'all')
```
And in case of "regular" external stylesheets they will be added to `$GLOBALS['TL_USER_CSS']`, which is then processed in `Controller::replaceDynamicScriptTags`.

fritzmg commented:
Also I just noticed the following: https://github.com/contao/contao/pull/42#issuecomment-431869814

fritzmg committed:
no tstamp fro non-combinable external stylesheets

fritzmg commented:
I have updated the PR and also removed the `$objFile->tstamp` for non-combinable external stylesheets. In order for this PR to work for external JavaScripts #42 needs to be reverted as discussed.

leofeyer commented:
Thank you @fritzmg.
